tools/checkpoint_conversion/convert_gpt2_checkpoints.py

```diff
@@ -0,0 +1,312 @@
+
+
+def define_tokenizer(
+    preset, num_params, hf_model_name, check_cloud_output=False
+):
+    print("\n-> Define the tokenizers.")
+    extract_dir = extract_dir = EXTRACT_DIR.format(num_params)
+    merges_path = os.path.join(extract_dir, "vocab.bpe")
+    vocab_path = os.path.join(extract_dir, "encoder.json")
+
+    keras_nlp_tokenizer = keras_nlp.models.GPT2Tokenizer(
+        vocabulary=vocab_path,
+        merges=merges_path,
+    )
+    hf_tokenizer = transformers.AutoTokenizer.from_pretrained(hf_model_name)
+
+    if not check_cloud_output:
+        print("\n-> Print MD5 checksum of the vocab files.")
+        print(f"`{vocab_path}` md5sum: ", get_md5_checksum(vocab_path))
+        print(f"`{merges_path}` md5sum: ", get_md5_checksum(merges_path))
+
+    return keras_nlp_tokenizer, hf_tokenizer
+
+
+def check_output(
+    preset,
+    keras_nlp_model,
+    keras_nlp_tokenizer,
+    hf_model,
+    hf_tokenizer,
+    check_cloud_output=False,
+):
+    print("\n-> Check the outputs.")
+    input_str = ["the quick brown fox ran, galloped and jumped."]
+
+    # KerasNLP
+    token_ids = keras_nlp_tokenizer(input_str)
+    padding_mask = token_ids != 0
+
+    keras_nlp_inputs = {
+        "token_ids": token_ids.to_tensor(),
+        "padding_mask": padding_mask.to_tensor(),
+    }
+    keras_nlp_output = keras_nlp_model.predict(keras_nlp_inputs)
+
+    # HF
+    hf_inputs = hf_tokenizer(input_str, return_tensors="pt")
+    hf_output = hf_model(**hf_inputs).last_hidden_state
+
+    print("KerasNLP output:", keras_nlp_output[0, 0, :10])
+    print("HF output:", hf_output[0, 0, :10])
+    print("Difference:", np.mean(keras_nlp_output - hf_output.detach().numpy()))
+
+    if check_cloud_output:
+        print("\n-> Check KerasNLP cloud output.")
+        keras_nlp_cloud_model = keras_nlp.models.GPT2.from_preset(
+            preset,
+            load_weights=True,
+        )
+        keras_nlp_cloud_output = keras_nlp_cloud_model.predict(keras_nlp_inputs)
+        print(
+            "Difference:",
+            tf.reduce_mean(keras_nlp_output - keras_nlp_cloud_output),
+        )
+    else:
+        # Show the MD5 checksum of the model weights.
+        print("Model md5sum: ", get_md5_checksum(f"./{preset}.h5"))
+
+    return keras_nlp_output
+
+
+def main(preset, check_cloud_output):
+    num_params = PRESET_MAP[preset][0]
+    hf_model_name = PRESET_MAP[preset][1]
+
+    download_model(preset, num_params)
+
+    keras_nlp_model = convert_checkpoints(preset, num_params)
+
+    print("\n-> Load HF model.")
+    hf_model = transformers.AutoModel.from_pretrained(hf_model_name)
+    hf_model.eval()
+
+    keras_nlp_tokenizer, hf_tokenizer = define_tokenizer(
+        preset, num_params, hf_model_name, check_cloud_output
+    )
+
+    check_output(
+        preset,
+        keras_nlp_model,
+        keras_nlp_tokenizer,
+        hf_model,
+        hf_tokenizer,
+        check_cloud_output,
+    )
+
+
+if __name__ == "__main__":
```

Comment:
Great script @abheesht17!

Given the style here I wonder if we should make a `CheckpointConversion` class with overridable methods for each of the functions defined here. For future scripts the result would be more structure, less boilerplate and a class that knows how to "run" itself. Thoughts @mattdangerw?

Reply:
After reading this over, I would probably go lighter weight for now, and maybe just move `get_md5_checksum` into a `checkpoint_utils.py` file in this directory.

Overall, having a base class sounds good. But I'm not sure the structure will look the same across all our models. E.g. distilbert is only huggingface, so there is no separate download step, you just use the huggingface model directly. Maybe we should look at the refactor after we have done this for another model type of two.